inc/mailcollector.class.php

```diff
@@ -786,6 +768,7 @@ function buildTicket($uid, $options = []) {
       // pattern: GLPI-{itemtype}-{items_id}
       // ex: GLPI-Ticket-26739
       $ref_match = "/GLPI-[A-Z]\w+-([0-9]+)/";
+      $ref_match = $this->getMessageMatch();
```

Comment:
```suggestion
```
The match on headers is made to find a match with string generated here: https://github.com/glpi-project/glpi/blob/9.4/bugfixes/inc/notificationeventmailing.class.php#L130.

The `$this->getMessageMatch()` is made to find ticket id in subject of the mail, so this is not the same pattern. Maybe it should be renamed to `getSubjectMatch`.

Reply:
this part will be reverted